packages/@aws-cdk/aws-apigateway/lib/access-log.ts

```diff
@@ -0,0 +1,556 @@
+  /**
+   * If this flag is enabled, $context.identity.caller will be output to the log
+   * @default true
+   */
+  caller?: boolean,
+  /**
+   * If this flag is enabled, $context.identity.user will be output to the log
+   * @default true
+   */
+  user?: boolean,
+  /**
+   * If this flag is enabled, $context.requestTime will be output to the log
+   * @default true
+   */
+  requestTime?: boolean,
+  /**
+   * If this flag is enabled, $context.httpMethod will be output to the log
+   * @default true
+   */
+  httpMethod?: boolean,
+  /**
+   * If this flag is enabled, $context.resourcePath will be output to the log
+   * @default true
+   */
+  resourcePath?: boolean,
+  /**
+   * If this flag is enabled, $context.status will be output to the log
+   * @default true
+   */
+  status?: boolean,
+  /**
+   * If this flag is enabled, $context.protocol will be output to the log
+   * @default true
+   */
+  protocol?: boolean,
+  /**
+   * If this flag is enabled, $context.responseLength will be output to the log
+   * @default true
+   */
+  responseLength?: boolean
+}
+
+/**
+ * factory methods for access log format.
+ */
+export class AccessLogFormat {
+  /**
+   * Generate Common Log Format.
+   */
+  public static clf(): IAccessLogFormat {
+    return {
+      format: `${AccessLogField.contextIdentitySourceIp()} ${AccessLogField.contextIdentityCaller()} ${AccessLogField.contextIdentityUser()} \
+[${AccessLogField.contextRequestTime()}] "${AccessLogField.contextHttpMethod()} ${AccessLogField.contextResourcePath()} ${AccessLogField.contextProtocol()}" \
+${AccessLogField.contextStatus()} ${AccessLogField.contextResponseLength()} ${AccessLogField.contextRequestId()}`
+    };
+  }
+
+  /**
+   * Generate standard json log format.
+   */
+  public static jsonWithStandardFields(
+    {
+      ip = true,
+      caller = true,
+      user = true,
+      requestTime = true,
+      httpMethod = true,
+      resourcePath= true,
+      status = true,
+      protocol = true,
+      responseLength = true
+    }: IJsonWithStandardFieldProps = {}): IAccessLogFormat {
+    return {
+      format: JSON.stringify({
+        requestId: AccessLogField.contextRequestId(),
+        ip: ip ? AccessLogField.contextIdentitySourceIp() : undefined,
+        caller: caller ? AccessLogField.contextIdentityCaller() : undefined,
+        user: user ? AccessLogField.contextIdentityUser() : undefined,
+        requestTime: requestTime ? AccessLogField.contextRequestTime() : undefined,
+        httpMethod: httpMethod ? AccessLogField.contextHttpMethod() : undefined,
+        resourcePath: resourcePath ? AccessLogField.contextResourcePath() : undefined,
+        status: status ? AccessLogField.contextStatus() : undefined,
+        protocol: protocol ? AccessLogField.contextProtocol() : undefined,
+        responseLength: responseLength ? AccessLogField.contextResponseLength() : undefined
+      })
+    };
+  }
+
+  /**
+   * Custom log format.
+   * @param format
```

Comment:
Expand a bit and briefly explain when to use this and how. An example would be useful.